Makefile.Common

```diff
@@ -74,3 +74,6 @@ fmt:
 scan:
 	$(OSV) -r .
 
+.PHONY: mdatagen
+mdatagen:
+	$(MDATAGEN) metadata.yaml
```

Comment:
I am not sure the history but compgen README referenced a makefile target for mdatagen. It is not there that I could find

Reply:
ok, this one is rather complicated. But the TL;DR is that I think we can revert this change.

Here's how it works:

* Run compgen to build a new component
* You now have a Makefile rendered from this [template](https://github.com/liatrio/liatrio-otel-collector/blob/main/cmd/compgen/cmd/receiver/templates/Makefile.tmpl) which imports the top-level Makefile.Common
* Makefile.Common defines the [gen](https://github.com/liatrio/liatrio-otel-collector/blob/main/Makefile.Common#L62-L63) target as `go generate ./...` which runs at the root dir of the new component
* `go generate ./...` executes mdatagen based on the comment in the generated [doc.go](https://github.com/liatrio/liatrio-otel-collector/blob/main/cmd/compgen/cmd/receiver/templates/doc.go.tmpl#L4) file

@jburns24 can we test this by generating a new dummy receiver and running `make gen` from the root of the new component's repo? As you already mentioned in our call, the relative path to the mdatagen binary is likely wrong. Nice find!!

Reply:
I'd opt for not adding back to the `mdatagen` rule target in the makefile, and fixing the doc instead.

Reply:
I thought I pushed up these changes 🤦